### ISSUE TITLE
 can‘t build pytorch v1.9.0 error:`error: namespace "thrust" has no member "host_vector"`

### ISSUE THREAD

**Issue by tanguofu** [pytorch/pytorch#64984]
## 🐛 Bug

```
/workspace/pytorch/caffe2/utils/math_gpu.cu(908): error: namespace "thrust" has no member "host_vector"

/workspace/pytorch/caffe2/utils/math_gpu.cu(908): error: expected an expression

/workspace/pytorch/caffe2/utils/math_gpu.cu(909): error: namespace "thrust" has no member "host_vector"

/workspace/pytorch/caffe2/utils/math_gpu.cu(909): error: expected an expression

/workspace/pytorch/caffe2/utils/math_gpu.cu(910): error: namespace "thrust" has no member "host_vector"
```

## To Reproduce

dockerfile:
```

FROM nvcr.io/nvidia/pytorch:21.08-py3

ARG TORCH_VERSION=v1.9.0

RUN echo "rebuild pytorch use nccl" && \
  echo 'export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0"' >> /root/.bashrc && \
  echo 'export FORCE_CUDA="1"' >> /root/.bashrc && \
  source /root/.bashrc && \
  git clone https://github.com/pytorch/pytorch.git && \
  cd pytorch && \
  echo "swich branch ${TORCH_VERSION} " && \
  git checkout ${TORCH_VERSION} && \
  git submodule update --init --recursive -f && \
  export USE_NINJA=OFF USE_SYSTEM_NCCL=1 NCCL_INCLUDE_DIR=/usr/local/cuda-11/targets/x86_64-linux/include  NCCL_LIB_DIR=/usr/local/cuda-11/targets/x86_64-linux/lib && \
  python3 setup.py bdist_wheel && \
  pip3 install -U dist/*.whl && \
  rm -fr /pytorch && \
  echo "done"
```

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

nvcr.io/nvidia/pytorch:21.08-py3 d

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): v1.9.0 
 - OS (e.g., Linux): ubuntu
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @malfet @seemethere @ngimel

**Comment by mruberry** [comment]
cc @malfet -- what's the right label for these issues? (should we create an "oncall: build" label?)

**Comment by malfet** [comment]
@mruberry I think `module: build` is the right one here, though it needs a more details to make it actionable.
Will add section on reporting build issues to https://github.com/pytorch/pytorch/blob/master/.github/ISSUE_TEMPLATE/bug-report.md

**Comment by ptrblck** [comment]
Related issue: https://github.com/pytorch/pytorch/issues/64484#issuecomment-912826851
tl;dr: use an older CUDA version, cherry-pick the fix into the `1.9.0` branch or build from master.

**Comment by ngimel** [comment]
Thanks @ptrblck! 
@tanguofu cuda 11.4 is not supported with pytorch 1.9
